File: Nodejs/backjoonNode/array/10813.js
/*
백준 10813: 공 바꾸기
https://www.acmicpc.net/problem/10813
*/

const fs = require('fs');
const input = fs.readFileSync("../inputt.txt").toString().trim().split("\r\n");

const [N, M] = input[0].split(" ").map(e => parseInt(e));
const MArr = input.slice(1).map(e => e.split(" ").map(el => parseInt(el)));
let Box = Array(N).fill().map((el, idx) => {return idx + 1});
let tmp;

for (let i = 0; i < M; i++) {
    if (MArr[i][0] != MArr[i][1]) {
        tmp = Box[MArr[i][0]-1];
        Box[MArr[i][0]-1] = Box[MArr[i][1]-1];
        Box[MArr[i][1]-1] = tmp;
    }
}

console.log(...Box);

File: C/백준/Step4/10813.c
# include <stdio.h>

int main() {
    int N, M;
    int ij[2];
    int tmp;

    scanf("%d %d", &N, &M);

    int Box[100]; // 실제로는 100대신 N을 대입하여 Array 구성

    for (int i = 0; i < N; i++) {
        Box[i] = i + 1;
    }

    for (int i = 0; i < M; i++) {
        scanf("%d %d", &ij[0], &ij[1]);

        if (ij[0] != ij[1]) {
            tmp = Box[ij[0] - 1];
            Box[ij[0] - 1] = Box[ij[1] - 1];
            Box[ij[1] - 1] = tmp;
        }
    }

    for (int i = 0; i < N; i++) {
        printf("%d ", Box[i]);
    }

    return 0;
}